bigbluebutton-html5/imports/ui/components/video-provider/auto-play-video/component.jsx

```diff
@@ -0,0 +1,119 @@
+import React, { Component } from 'react';
+import logger from '/imports/startup/client/logger';
+
+const CAMERA_PROFILES = Meteor.settings.public.kurento.cameraProfiles;
+
+class AutoPlayVideo extends Component {
```

Comment:
I might be missing something here, but why is the video-preview webcam enumeration and picking code being duplicated here? If the idea is to just fetch whatever gUM returns, this isn't needed at all (and adds duplicated code of a domain that's already a bit unstable).
If I got it right, we could just do things directly in video-provider without this component. At video-provider's `getCameraProfile`, if there's no default webcam set in the `Session` storage, leaving the `constraints.deviceId` member undefined and moving the default profile fetching to there if there's none in the storage should give the same behavior as this.

Reply:
The duplication also leads to issues with maintaining the different paths. There's no surfacing of errors and already you can look at the code in this PR and see that it's missing the overriding of the default profile.

It would be easier to maintain if we could instead leverage the existing video preview modal and have it automatically publish once it's found a webcam instead of showing the preview.

Reply:
Yeah, I second the approach of re-using video-preview, even better than shoving it to video-provider. Only drawback I think is using two gUMs. But that's a problem that is already a thing today and we should fix out of the scope of this PR. Either way, both solutions would be simpler and both would help surfacing errors since video-provider also handles gUM-related errors with toast messages.

Reply:
I hadn't thought about the double gUM not being needed anymore. We can just save the stream from the first one and pass it to the preview video directly.

Reply:
I meant the other gUM, the one video-provider uses. We pass the `deviceId` to it, not the stream, hence it also runs a gUM. We should pass the stream to it, the utils library we use support it. But yeah, the enumeration gUM would also be unneeded.